pennylane/ops/qutrit/observables.py

```diff
@@ -110,13 +110,18 @@ def eigendecomposition(self):
                 Hermitian observable
         """
         Hmat = self.matrix()
-        Hmat = qml.math.to_numpy(Hmat)
-        Hkey = tuple(Hmat.flatten().tolist())
-        if Hkey not in THermitian._eigs:
-            w, U = np.linalg.eigh(Hmat)
-            THermitian._eigs[Hkey] = {"eigvec": U, "eigval": w}
 
-        return THermitian._eigs[Hkey]
+        if not qml.math.is_abstract(Hmat):
+            Hmat = qml.math.to_numpy(Hmat)
+            Hkey = tuple(Hmat.flatten().tolist())
+            if Hkey not in THermitian._eigs:
+                w, U = qml.math.eigh(Hmat)
```

Comment:
How come `qml.math.linalg.eigh` doesn't work? 🤔

Reply:
Apparently it does, so I'm removing it from multi dispatch.